node/src/components/consensus/highway_core/state/index_panorama.rs

```diff
@@ -38,15 +39,15 @@ impl IndexPanorama {
         state: &'a State<C>,
     ) -> Self {
         let mut validator_map: ValidatorMap<IndexObservation> =
-            ValidatorMap::from(vec![IndexObservation::None; panorama.len()]);
+            ValidatorMap::from(vec![IndexObservation::Count(0); panorama.len()]);
         for (vid, obs) in panorama.enumerate() {
             let index_obs = match obs {
-                Observation::None => IndexObservation::None,
-                Observation::Correct(hash) => state
-                    .maybe_unit(hash)
-                    .map_or(IndexObservation::None, |unit| {
-                        IndexObservation::Correct(unit.seq_number)
-                    }),
+                Observation::None => IndexObservation::Count(0),
```

Comment:
If `IndexObservation::Count` is supposed to represent the first sequence number we're missing then this one should be `Count(1)`, right? We don't have units with sequence number 0.